doc/rest-authentication.md

```diff
@@ -0,0 +1,106 @@
+# Glusterd2 REST API authentication
+
+Glusterd2 REST API authentication is based on [JWT](https://jwt.io). If REST
+authentication is enabled then each client request should include
+`Authorization` header as example below.
+
+    Authorization: bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnbHVzdGVyY2xpIiwiaWF0IjoxNTMxNzI5NzQyLCJleHAiOjE1MzE3Mjk3NTJ9._WIwO7PrHUHIT62SdzfkyNjqD1GEgX2cYqN8ACZCtaw
+
+**Note**: REST Authentication can be disabled by adding `restauth=false`
+in Glusterd2 config file(Default path is
+`/var/lib/glusterd2/glusterd2.toml` in case of rpm installation)
+
+## Default user
+
+When `glusterd2` starts for the first time, it creates the
+`$GLUSTERD2_STATEDIR/auth` file which will contain the secret. If
+"gluster" user group is available in the system then this `auth` file
+can be read by any user in that machine who are part of "gluster"
+group.
+
+**Note**: If "gluster" user group is not available during first start
+of `glusterd2` it limits the read permission to `root:root`
+
+## glustercli
+
+With default installation, `glustercli` will know the location of
```

Comment:
Perhaps, "The glustercli command knows the default location of the auth file." would read more clearly.